### ISSUE TITLE
Agent Groups

### ISSUE THREAD

**Issue by damies13** [damies13/rfswarm#62]
**Is your feature request related to a problem? Please describe.**
- when testing an application with some external and some internal agents
- when testing with a mix of robot libraries and some agent machines only support one library type.
- having a mix of agent os's, i.e. want the SeleniumLibrary scripts to run on any but the WhiteLibrary scripts should only be assigned to windows agents

**Describe the solution you'd like**

Ability to assign Agent Groups and select which scripts run on which groups

**Comment by damies13** [comment]
This code could be used to identify which robotframework Libraries are installed, potentially this could be used to identify which agents to send which robot scripts to.

```
import pkg_resources
for i in installed_packages:
     if i.key.startswith("robotframework"):
             print(i.key)


robotframework-browser
robotframework
robotframework-sshlibrary
robotframework-sikulilibrary
robotframework-seleniumlibrary
robotframework-requests
robotframework-pythonlibcore
robotframework-jsonvalidator

```

**Comment by damies13** [comment]
Finished implementing this feature, it's taken nearly a week to implement just this feature, but hopefully it's worth the wait.
- the agent will detect robotframework libraries and list them as a property of that agent
- the agent will detect the os and rfswarm version and list them as a property of that agent
- the agent will read custom (user defined) properties from the ini file or the command line
- the gui server now has a script setting screen that can be selected from the script row
- the script setting screen allows you to specify which libraries should be excluded from displaying results
- the script setting screen allows you to specify command line options you might want to pass to robot.exe (e.g. -M -G -v)
- the script setting screen allows you to create filter rules for requiring or excluding agents with specific properties

This change has been merged into release 0.6.3
